test/models/userProfileModelManager.test.js

```diff
@@ -0,0 +1,180 @@
+        userProfileMock.lastname
+      );
+      userProfileMock.userId = newUser.id;
+      const newUserProfile = await UserProfileModelManager.createUserProfile(
+        newUser.id,
+        userProfileMock.avatar,
+        userProfileMock.bio,
+        userProfileMock.username,
+        userProfileMock.country,
+        userProfileMock.phonenumber,
+        userProfileMock.gender,
+        userProfileMock.emailnotification
+      );
+      newModels.newUser = newUser;
+      newModels.newUserProfile = newUserProfile;
+      newModels.user = await User.findOne({
+        where: { id: newUser.id },
+        include: [{ model: userprofile, as: 'userprofile' }]
+      });
+    });
+    it('should have new user profile created', () => {
+      const {
+        avatar,
+        userId,
+        bio,
+        username,
+        country,
+        phonenumber,
+        gender,
+        emailnotification
+      } = userProfileMock;
+      expect(newModels.user.userprofile.avatar).to.equal(avatar);
+      expect(newModels.user.userprofile.UserId).to.equal(userId);
+      expect(newModels.user.userprofile.bio).to.equal(bio);
+      expect(newModels.user.userprofile.username).to.equal(username);
+      expect(newModels.user.userprofile.country).to.equal(country);
+      expect(newModels.user.userprofile.phonenumber).to.equal(phonenumber);
+      expect(newModels.user.userprofile.gender).to.equal(gender);
+      expect(newModels.user.userprofile.emailnotification).to.equal(emailnotification);
+    });
+  });
+  describe('Update user profile', async () => {
+    before(async () => {
+      const updatedUserProfileMock = {
+        avatar: 'https://testavarturl2.com',
+        bio: 'Test bio2',
+        username: 'user12',
+        country: 'Ghana',
+        phonenumber: '+2348126573983',
+        gender: 'female',
+        emailnotification: true
+      };
+      newModels.updatedUserProfileMock = updatedUserProfileMock;
+      const newUser = await UserModelManager.create(
+        userProfileMock.email,
+        userProfileMock.password,
+        userProfileMock.firstname,
+        userProfileMock.lastname
+      );
+      userProfileMock.userId = newUser.id;
+      const newUserProfile = await UserProfileModelManager.createUserProfile(
+        newUser.id,
+        userProfileMock.avatar,
+        userProfileMock.bio,
+        userProfileMock.username,
+        userProfileMock.country,
+        userProfileMock.phonenumber,
+        userProfileMock.gender,
+        userProfileMock.emailnotification
+      );
+      newModels.newUser = newUser;
+      newModels.newUserProfile = newUserProfile;
+      const {
+        phonenumber,
+        avatar,
+        bio,
+        username,
+        country,
+        gender,
+        emailnotification
+      } = updatedUserProfileMock;
+      await UserProfileModelManager.updateUserProfile(newUser.id, {
+        phonenumber
+      });
+      await UserProfileModelManager.updateUserProfile(newUser.id, {
```

Comment:
I don't think these multiple updates are necessary. You can update all the fields in one go.